File: mario.c
#include<cs50.h>
#include<stdio.h>

int main(void)
{
    int n;
    int i;
    int j;
    int x;
    
    do
    {
        //asks for height
        n = get_int("please enter a hight between 1 and 8:");
    }
    //sorts and rejects input
    while (n <= 0 || n >= 9);
    
    for (i = 1; i < n + 1; i++)
    {
        //corrects the spacing
        for (x = (n - i); x > 0; x--)
        {
            printf(" ");
                
        }
        for (j = 0; j < i; j++)
        {
            printf("#");
        }   
        printf("\n");
    }
}

File: mario.py
from cs50 import get_int
while True:
    height = get_int("please enter a hight between 1 and 8:")
    if height <= 0 or height > 8:
        continue   
    
    for i in range(1, height + 1):
            num_hashes = i
            num_spaces = height - num_hashes
            print(" " * num_spaces, end="")
            print("#" * num_hashes)
    break
